src/main/java/org/elasticsearch/search/action/SearchServiceTransportAction.java

```diff
@@ -430,134 +348,78 @@ public void sendExecuteFetchScroll(DiscoveryNode node, final ShardFetchRequest r
-            }, listener);
-        } else {
-            transportService.sendRequest(node, action, request, new BaseTransportResponseHandler<FetchSearchResult>() {
-
-                @Override
-                public FetchSearchResult newInstance() {
-                    return new FetchSearchResult();
-                }
-
-                @Override
-                public void handleResponse(FetchSearchResult response) {
-                    listener.onResult(response);
-                }
-
-                @Override
-                public void handleException(TransportException exp) {
-                    listener.onFailure(exp);
-                }
-
-                @Override
-                public String executor() {
-                    return ThreadPool.Names.SAME;
-                }
-            });
-        }
+        transportService.sendRequest(node, action, request, new BaseTransportResponseHandler<FetchSearchResult>() {
+
+            @Override
+            public FetchSearchResult newInstance() {
+                return new FetchSearchResult();
+            }
+
+            @Override
+            public void handleResponse(FetchSearchResult response) {
+                listener.onResult(response);
+            }
+
+            @Override
+            public void handleException(TransportException exp) {
+                listener.onFailure(exp);
+            }
+
+            @Override
+            public String executor() {
+                return ThreadPool.Names.SAME;
+            }
+        });
     }
 
     public void sendExecuteScan(DiscoveryNode node, final ShardSearchTransportRequest request, final SearchServiceListener<QuerySearchResult> listener) {
-        if (clusterService.state().nodes().localNodeId().equals(node.id())) {
-            execute(new Callable<QuerySearchResult>() {
-                @Override
-                public QuerySearchResult call() throws Exception {
-                    return searchService.executeScan(request);
-                }
-            }, listener);
-        } else {
-            transportService.sendRequest(node, SCAN_ACTION_NAME, request, new BaseTransportResponseHandler<QuerySearchResult>() {
-
-                @Override
-                public QuerySearchResult newInstance() {
-                    return new QuerySearchResult();
-                }
-
-                @Override
-                public void handleResponse(QuerySearchResult response) {
-                    listener.onResult(response);
-                }
-
-                @Override
-                public void handleException(TransportException exp) {
-                    listener.onFailure(exp);
-                }
-
-                @Override
-                public String executor() {
-                    return ThreadPool.Names.SAME;
-                }
-            });
-        }
+        transportService.sendRequest(node, SCAN_ACTION_NAME, request, new BaseTransportResponseHandler<QuerySearchResult>() {
```

Comment:
these handlers all look the same can we have a new abstract base class where we pass the handler and the threadpool as the ctor arg  and only override the newInstance method?